src/api/mob.js

```diff
@@ -0,0 +1,81 @@
+import express from 'express';
+
+export default (dispatch, action, storage) => {
+  const router = new express.Router();
+
+  const getTimer = (timerId) => storage.read()[timerId];
+
+  router.get('/add/:name', (request, response) => {
+    const { name } = request.params;
+    const { mob, lockedMob } = getTimer(request.timerId);
+
+    if (mob.includes(name)) {
+      return response
+        .status(400)
+        .json({ message: 'User is already in mob' })
+        .end();
+    }
+
+    if (lockedMob) {
+      return response
+        .status(400)
+        .json({ message: 'Mob is locked' })
+        .end();
+    }
+
+    dispatch(action.AddUser(name, request.timerId));
+
+    return response.status(201).end();
+  });
+
+  router.get('/remove/:name', (request, response) => {
+    const { name } = request.params;
+    const { lockedMob } = getTimer(request.timerId);
+
+    if (lockedMob) {
+      return response
+        .status(400)
+        .json({ message: 'Mob is locked' })
+        .end();
+    }
+
+    dispatch(action.RemoveUser(name, request.timerId));
+
+    return response.status(201).end();
+  });
+
+  router.get('/cycle', (request, response) => {
+    dispatch(action.CycleMob(request.timerId));
+
+    return response.status(204).end();
+  });
+
+  router.get('/shuffle', (request, response) => {
+    const { lockedMob } = getTimer(request.timerId);
+
+    if (lockedMob) {
+      return response
+        .status(400)
+        .json({ message: 'Mob is locked' })
+        .end();
+    }
+
+    dispatch(action.ShuffleMob(request.timerId));
+
+    return response.status(204).end();
+  });
+
+  router.get('/lock', (request, response) => {
```

Comment:
Originally lock/unlock was for calculating end-of-cycle, but I'd rather track a rotation counter and check of `numberOfRotations > 0 && numberOfRotations % mob.length === 0` to know if the mob is at end of cycle.

This would remove the mob lock/unlock, but I wouldn't mind having a separate locking mechanism to prevent people from randomly joining, like whereby's knock or a password.

Reply:
See #43